solitaire.rb

```diff
@@ -1,2 +1,154 @@
+require_relative 'keystreamgenerator'
+
 class Solitaire
+
+  def initialize(opts={})
+    @message           = opts[:message]
+    @encrypted_message = opts[:encrypted_message]
+  end
+
+  def encrypt
+    sanitize_message
+    convert_message_to_numbers
+    convert_keystream_to_numbers
+    add_message_numbers
+    convert_to_letters(added_message_numbers)
+  end
+
+  def decrypt
+    convert_encrypted_message_to_numbers
+    convert_keystream_to_numbers
+    subtract_message_numbers
+    convert_to_letters(subtracted_message_numbers)
+  end
+
+  private
+
+  LETTER_CONVERTER = Hash[('A'..'Z').zip (1..26)]
+  NUMBER_CONVERTER = Hash[(1..26).zip ('A'..'Z')]
+
+  attr_reader(
+    :message,
+    :encrypted_message,
+    :keystream_length,
+    :keystream_letters,
+    :sanitized_message,
+    :converted_message,
+    :converted_keystream,
+    :added_message_numbers,
+    :cleaned_message,
+    :grouped_message,
+    :x_added,
+    :converted_encrypted_message,
+    :subtracted_message_numbers,
+  )
```

Comment:
I've removed some attr_readers I should be able to clean this up a bit more when extract another class or two here.